modules/web/middleware/cookie.go

```diff
@@ -48,7 +48,23 @@ func SetSiteCookie(resp http.ResponseWriter, name, value string, maxAge int) {
 	if maxAge < 0 {
 		// There was a bug in "setting.SessionConfig.CookiePath" code, the old default value of it was empty "".
 		// So we have to delete the cookie on path="" again, because some old code leaves cookies on path="".
-		cookie.Path = strings.TrimSuffix(setting.SessionConfig.CookiePath, "/")
-		resp.Header().Add("Set-Cookie", cookie.String())
+		// The code was updated, but it behaves differently depending on the
+		// value of AppSubURL.  When AppSubURL is non-empty, the cookie with a
+		// trailing slash must be deleted.
+		withoutTrailingSlash := strings.TrimSuffix(setting.SessionConfig.CookiePath, "/")
+		withTrailingSlash := ensureSuffix(setting.SessionConfig.CookiePath, "/")
+		for _, path := range []string{withoutTrailingSlash, withTrailingSlash} {
+			if setting.SessionConfig.CookiePath != path {
+				cookie.Path = path
+				resp.Header().Add("Set-Cookie", cookie.String())
+			}
```

Comment:
```suggestion
		cookie.Path = path
		resp.Header().Add("Set-Cookie", cookie.String())
```

Reply:
I mean:

```
if maxAge < 0 {
   for {
      resp.Header().Add("Set-Cookie", cookie.String())
   }
} else {
  resp.Header().Add("Set-Cookie", cookie.String())
}
```

Reply:
I'm not sure how the line could be moved while also deleting the condition. `setting.SessionConfig.CookiePath` may or may not end with a trailing slash, depending on the value of `AppSubURL`. I updated it to not duplicate `resp.Header().Add("Set-Cookie", cookie.String())`, so hopefully this is what you mean.